### ISSUE TITLE
VRF Sizing Problem and PLR error

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#6118]
Issue overview
--------------
User file with VRF has some problems:

1.  Hard flow rates specified in the VRF terminal unit do not get used in sizing the VRF coils - seems that they should.
2.  ~~With the terminal unit flow rates autosized, the simulation throws errors like this:~~ (in v8.6, not in v8.7)
```
   ** Warning ** ZoneHVAC:TerminalUnit:VariableRefrigerantFlow "THERMAL ZONE 1-6|VTU-1"
   **   ~~~   ** Terminal unit part-load ratio calculation failed: PLR limits of 0 to 1 exceeded
   **   ~~~   ** Please fill out a bug report and forward to the EnergyPlus support group.
```
3. And then it fails with a fatal error:
```
   ** Severe  ** PsyRhoAirFnPbTdbW: RhoAir (Density of Air) is calculated <= 0 [-410.64918].
   **   ~~~   ** pb =[99700.00], tdb=[-274.00], w=[9.9888827E-006].
   **   ~~~   **  Routine=Unknown, Environment=ANNUAL RUN FOR 2013, at Simulation time=09/05 07:00 - 07:01
   **  Fatal  ** Program terminates due to preceding condition.
```

### Details
Some additional details for this issue (if relevant):
 - Platform Win64
 - Version of EnergyPlus v8.7.0
 - Helpdesk ticket number 11995

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added to DevSupport\DefectFiles.  Use USA_CA_San.Diego-Montgomery.Field.722903_TMY3.epw
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by rraustad** [comment]
So what should we do here? Limit OA flow to max of cooling/heating? Raise max cooling/heating to allow user specified OA? Let sizing run as is and limit OA flow in calc routine? I guess it makes most sense to figure this out in the sizing routine so user can be warned of the change. I am leaning towards limiting OA flow to max of cooling/heating and if not autosized, warn user.

    ZoneHVAC:TerminalUnit:VariableRefrigerantFlow,
      Thermal Zone 1-3|VTU-1,  !- Zone Terminal Unit Name
      autosize,                !- Cooling Supply Air Flow Rate {m3/s}
      0,                       !- No Cooling Supply Air Flow Rate {m3/s}
      autosize,                !- Heating Supply Air Flow Rate {m3/s}
      0,                       !- No Heating Supply Air Flow Rate {m3/s}
      0.155261915998112,       !- Cooling Outdoor Air Flow Rate {m3/s}
      0.155261915998112,       !- Heating Outdoor Air Flow Rate {m3/s}

    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, Design Size Cooling Supply Air Flow Rate [m3/s], 0.10443
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, Design Size Heating Supply Air Flow Rate [m3/s], 0.10443
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, User-Specified No Cooling Supply Air Flow Rate [m3/s], 0.00000
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, User-Specified No Heating Supply Air Flow Rate [m3/s], 0.00000
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, Design Size Outdoor Air Flow Rate During Cooling Operation [m3/s], 4.70000E-002
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, User-Specified Outdoor Air Flow Rate During Cooling Operation [m3/s], 0.15526
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, Design Size Outdoor Air Flow Rate During Heating Operation [m3/s], 4.70000E-002
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, User-Specified Outdoor Air Flow Rate During Heating Operation [m3/s], 0.15526

    Component Sizing Information, Fan:ConstantVolume, THERMAL ZONE 1-3|VTU-1|SMPL_SUPPLYCONSTVOL, Design Size Maximum Flow Rate [m3/s], 0.10443

**Comment by mjwitte** [comment]
That's really a side issue.  I should have changed those to autosize.  But I would agree that there should be a warning (once) that the user-specified OA flow is greater than supply flow and will be limited to supply flow.  On the other hand, if these were autosized and the autosized OA flow came out greater, then the supplies would be increased to match.  So, maybe that's what should happen here.

**Comment by rraustad** [comment]
New warnings based on previous discussion:

New warning:

    ************* SizeVRF: Potential issue with equipment sizing for ZoneHVAC:TerminalUnit:VariableRefrigerantFlow THERMAL ZONE 1-3|VTU-1
    **   ~~~   ** User-Specified Outdoor Air Flow Rate During Cooling Operation of 0.15526 [m3/s]
    **   ~~~   ** is greater than maximum Air Flow Rate During Cooling Operation of 0.10443 [m3/s]
    **   ~~~   ** User-Specified Outdoor Air Flow Rate During Cooling Operation will be reset to 0.10443 [m3/s] and the simulation continues

and now existing warning is based on the reset value:

    ************* SizeVRF: Potential issue with equipment sizing for ZoneHVAC:TerminalUnit:VariableRefrigerantFlow THERMAL ZONE 1-3|VTU-1
    **   ~~~   ** User-Specified Outdoor Air Flow Rate During Cooling Operation of 0.10443 [m3/s]
    **   ~~~   ** differs from Design Size Outdoor Air Flow Rate During Cooling Operation of 4.70000E-002 [m3/s]
    **   ~~~   ** This may, or may not, indicate mismatched component sizes.
    **   ~~~   ** Verify that the value entered is intended and is consistent with other components.

New warning:

    ************* SizeVRF: Potential issue with equipment sizing for ZoneHVAC:TerminalUnit:VariableRefrigerantFlow THERMAL ZONE 1-3|VTU-1
    **   ~~~   ** User-Specified Outdoor Air Flow Rate During Heating Operation of 0.15526 [m3/s]
    **   ~~~   ** is greater than maximum Air Flow Rate During Heating Operation of 0.10443 [m3/s]
    **   ~~~   ** User-Specified Outdoor Air Flow Rate During Heating Operation will be reset to 0.10443 [m3/s] and the simulation continues

and now existing warning is based on the reset value:

    ************* SizeVRF: Potential issue with equipment sizing for ZoneHVAC:TerminalUnit:VariableRefrigerantFlow THERMAL ZONE 1-3|VTU-1
    **   ~~~   ** User-Specified Outdoor Air Flow Rate During Heating Operation of 0.10443 [m3/s]
    **   ~~~   ** differs from Design Size Outdoor Air Flow Rate During Heating Operation of 4.70000E-002 [m3/s]
    **   ~~~   ** This may, or may not, indicate mismatched component sizes.
    **   ~~~   ** Verify that the value entered is intended and is consistent with other components.

**Comment by rraustad** [comment]
Comparing err file from V8.7 to develop:

V8.7:

    ************* SizeVRF: Potential issue with equipment sizing for ZoneHVAC:TerminalUnit:VariableRefrigerantFlow THERMAL ZONE 1-3|VTU-1
    **   ~~~   ** User-Specified Outdoor Air Flow Rate During Cooling Operation of 0.15526 [m3/s]
    **   ~~~   ** differs from Design Size Outdoor Air Flow Rate During Cooling Operation of 4.70000E-002 [m3/s]
    **   ~~~   ** This may, or may not, indicate mismatched component sizes.
    **   ~~~   ** Verify that the value entered is intended and is consistent with other components.
    ************* SizeVRF: Potential issue with equipment sizing for ZoneHVAC:TerminalUnit:VariableRefrigerantFlow THERMAL ZONE 1-3|VTU-1
    **   ~~~   ** User-Specified Outdoor Air Flow Rate During Heating Operation of 0.15526 [m3/s]
    **   ~~~   ** differs from Design Size Outdoor Air Flow Rate During Heating Operation of 4.70000E-002 [m3/s]
    **   ~~~   ** This may, or may not, indicate mismatched component sizes.
    **   ~~~   ** Verify that the value entered is intended and is consistent with other components.
    ** Warning ** SizeDXCoil Coil:Cooling:DX:VariableRefrigerantFlow THERMAL ZONE 1-3|VTU-1|VRF COOLING COIL
    **   ~~~   ** ...Gross Rated Total Cooling Capacity [W] will be limited by the maximum rated volume flow per rated total capacity ratio.
    **   ~~~   ** ...DX coil volume flow rate ( m3/s ) = 0.104425
    **   ~~~   ** ...Requested capacity ( W ) = 90.917
    **   ~~~   ** ...Requested flow/capacity ratio ( m3/s/W ) = 1.148E-003
    **   ~~~   ** ...Maximum flow/capacity ratio ( m3/s/W ) = 6.041E-005
    **   ~~~   ** ...Adjusted capacity ( W ) = 1728.607

Develop:

    ************* SizeVRF: Potential issue with equipment sizing for ZoneHVAC:TerminalUnit:VariableRefrigerantFlow THERMAL ZONE 1-3|VTU-1
    **   ~~~   ** User-Specified Outdoor Air Flow Rate During Cooling Operation of 0.15526 [m3/s]
    **   ~~~   ** is greater than maximum Air Flow Rate During Cooling Operation of 0.10443 [m3/s]
    **   ~~~   ** User-Specified Outdoor Air Flow Rate During Cooling Operation will be reset to 0.10443 [m3/s] and the simulation continues
    ************* SizeVRF: Potential issue with equipment sizing for ZoneHVAC:TerminalUnit:VariableRefrigerantFlow THERMAL ZONE 1-3|VTU-1
    **   ~~~   ** User-Specified Outdoor Air Flow Rate During Cooling Operation of 0.10443 [m3/s]
    **   ~~~   ** differs from Design Size Outdoor Air Flow Rate During Cooling Operation of 4.70000E-002 [m3/s]
    **   ~~~   ** This may, or may not, indicate mismatched component sizes.
    **   ~~~   ** Verify that the value entered is intended and is consistent with other components.
    ************* SizeVRF: Potential issue with equipment sizing for ZoneHVAC:TerminalUnit:VariableRefrigerantFlow THERMAL ZONE 1-3|VTU-1
    **   ~~~   ** User-Specified Outdoor Air Flow Rate During Heating Operation of 0.15526 [m3/s]
    **   ~~~   ** is greater than maximum Air Flow Rate During Heating Operation of 0.10443 [m3/s]
    **   ~~~   ** User-Specified Outdoor Air Flow Rate During Heating Operation will be reset to 0.10443 [m3/s] and the simulation continues
    ************* SizeVRF: Potential issue with equipment sizing for ZoneHVAC:TerminalUnit:VariableRefrigerantFlow THERMAL ZONE 1-3|VTU-1
    **   ~~~   ** User-Specified Outdoor Air Flow Rate During Heating Operation of 0.10443 [m3/s]
    **   ~~~   ** differs from Design Size Outdoor Air Flow Rate During Heating Operation of 4.70000E-002 [m3/s]
    **   ~~~   ** This may, or may not, indicate mismatched component sizes.
    **   ~~~   ** Verify that the value entered is intended and is consistent with other components.
    ** Warning ** SizeDXCoil Coil:Cooling:DX:VariableRefrigerantFlow THERMAL ZONE 1-3|VTU-1|VRF COOLING COIL
    **   ~~~   ** ...Gross Rated Total Cooling Capacity [W] will be limited by the maximum rated volume flow per rated total capacity ratio.
    **   ~~~   ** ...DX coil volume flow rate ( m3/s ) = 0.104425
    **   ~~~   ** ...Requested capacity ( W ) = 90.917
    **   ~~~   ** ...Requested flow/capacity ratio ( m3/s/W ) = 1.148E-003
    **   ~~~   ** ...Maximum flow/capacity ratio ( m3/s/W ) = 6.041E-005
    **   ~~~   ** ...Adjusted capacity ( W ) = 1728.607

**Comment by mjwitte** [comment]
So, with the OA reduced, do the bad PLR and fatal psych errors still happen?

**Comment by rraustad** [comment]
I ran in V8.7 with no PLR warnings (using SanDiegoMont weather). The psyc warnings did show in V8.7. The only thing I changed was reducing the max number of warm up days from 100 to 15. I did see the issue above and tried a fix. I'll try again to get the PLR warnings to report. I'll run your defect file2 directly in V8.7.

**Comment by rraustad** [comment]
Note limits on performance curve min/max output below. At the inlet air mass flow rate, a 23 multiplier on capacity will push SAT to sub-zero temperatures (-272 C first time warning is tripped).

TotCapTempModFac = 23  = CurveValue( DXCoil( DXCoilNum ).CCapFTemp( Mode ), InletAirWetBulbC, CondInletTemp );

    Coil:Cooling:DX:VariableRefrigerantFlow,
     Thermal Zone 1-3|VTU-1|VRF Cooling Coil,  !- Name
     VRFCoolEIRFT,            !- Cooling Capacity Ratio Modifier Function of Temperature Curve Name
     VRFCPLFFPLR,             !- Cooling Capacity Modifier Curve Function of Flow Fraction Name

    Curve:Biquadratic,
     VRFCoolEIRFT,            !- Name
     0.989010541,             !- Coefficient1 Constant
     -0.02347967,             !- Coefficient2 x
     0.000199711,             !- Coefficient3 x**2
     0.005968336,             !- Coefficient4 y
     -1.0289E-07,             !- Coefficient5 y**2
     -0.00015686,             !- Coefficient6 x*y
     15,                      !- Minimum Value of x
     24,                      !- Maximum Value of x
     -5,                      !- Minimum Value of y
     23,                      !- Maximum Value of y
     23,                      !- Minimum Curve Output
     23,                      !- Maximum Curve Output

**Comment by rraustad** [comment]
There is another curve for CapFT in the input so the user simply chose the wrong curve for the VRF TU.

  Curve:Biquadratic,
    VRFCoolCapFT,            !- Name
    0.576856477,             !- Coefficient1 Constant
    0.017450222,             !- Coefficient2 x
    0.00058326,              !- Coefficient3 x**2
    1.15563E-07,             !- Coefficient4 y
    7.61504E-09,             !- Coefficient5 y**2
    -1.1503E-08,             !- Coefficient6 x*y
    15,                      !- Minimum Value of x
    24,                      !- Maximum Value of x
    -5,                      !- Minimum Value of y
    23,                      !- Maximum Value of y
    23,                      !- Minimum Curve Output
    23,                      !- Maximum Curve Output

**Comment by rraustad** [comment]
I'm running a corrected file now. Also, I see these existing warnings so the OA vs SA issue has been addressed in the VRF model in Init.

     ** Warning ** InitVRF: VRF Terminal Unit = [ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, "THERMAL ZONE 1-3|VTU-1"]
     **   ~~~   ** ...The Outdoor Air Flow Rate During Cooling Operation exceeds the Supply Air Flow Rate During Cooling Operation.
     **   ~~~   ** ...Outdoor Air Flow Rate During Cooling Operation = 0.1553 m3/s
     **   ~~~   ** ... Supply Air Flow Rate During Cooling Operation = 0.1044 m3/s
     **   ~~~   ** ...the outdoor air flow rate will be reduced to match and the simulation continues.

**Comment by rraustad** [comment]
Just sent a reply to HelpDesk Ticket 11995 to correct the users problems.

**Comment by rraustad** [comment]
Also, I don't see the word failed in the error file running in-Fixed2.idf.

**Comment by mjwitte** [comment]
Oh - that was happening in v8.6, but I see it's not in v8.7.

**Comment by rraustad** [comment]
See PR #6121

**Comment by mjwitte** [comment]
Ok, but what about item 1 above, the coil sizing when there's a hard flow rate in the terminal unit?

**Comment by rraustad** [comment]
I'll look at this. It's supposed to work given this is set:

    if ( CurZoneEqNum > 0 ) {
        ZoneEqSizing( CurZoneEqNum ).AirVolFlow = max( VRFTU( VRFTUNum ).MaxCoolAirVolFlow, VRFTU( VRFTUNum ).MaxHeatAirVolFlow );
    }

but only if the coil sizing is using this value correctly.

**Comment by rraustad** [comment]
Getting closer, the only thing I see now is that the heating coil did not pick up the user specified air flow rate. This could be a can of worms trying to fix this!

    ZoneHVAC:TerminalUnit:VariableRefrigerantFlow,
     Thermal Zone 1-3|VTU-1,  !- Zone Terminal Unit Name
     VRFCondAvailSched_2014,  !- Terminal Unit Availability Schedule
     Thermal Zone 1-3|VTU-1 ZoneExhaustAirDemandOutletNode,  !- Terminal Unit Air Inlet Node Name
     Thermal Zone 1-3|VTU-1 ZoneAirDemandInletNode,  !- Terminal Unit Air Outlet Node Name
     1.3,                !- Cooling Supply Air Flow Rate {m3/s}
     0.3,                !- No Cooling Supply Air Flow Rate {m3/s}
     1.2,                !- Heating Supply Air Flow Rate {m3/s}
     0.2,                !- No Heating Supply Air Flow Rate {m3/s}
     0.17,              !- Cooling Outdoor Air Flow Rate {m3/s}
     0.16,              !- Heating Outdoor Air Flow Rate {m3/s}
     0.1,                !- No Load Outdoor Air Flow Rate {m3/s}

    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, Design Size Cooling Supply Air Flow Rate [m3/s], 8.77703E-002
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, User-Specified Cooling Supply Air Flow Rate [m3/s], 1.30000
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, Design Size Heating Supply Air Flow Rate [m3/s], 8.77703E-002
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, User-Specified Heating Supply Air Flow Rate [m3/s], 1.20000
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, Design Size No Cooling Supply Air Flow Rate [m3/s], 8.77703E-002
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, User-Specified No Cooling Supply Air Flow Rate [m3/s], 0.30000
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, Design Size No Heating Supply Air Flow Rate [m3/s], 8.77703E-002
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, User-Specified No Heating Supply Air Flow Rate [m3/s], 0.20000
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, Design Size Outdoor Air Flow Rate During Cooling Operation [m3/s], 4.70000E-002
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, User-Specified Outdoor Air Flow Rate During Cooling Operation [m3/s], 0.17000
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, Design Size Outdoor Air Flow Rate During Heating Operation [m3/s], 4.70000E-002
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, User-Specified Outdoor Air Flow Rate During Heating Operation [m3/s], 0.16000
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, Design Size Outdoor Air Flow Rate When No Cooling or Heating is Needed [m3/s], 4.70000E-002
    Component Sizing Information, ZoneHVAC:TerminalUnit:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1, User-Specified Outdoor Air Flow Rate When No Cooling or Heating is Needed [m3/s], 0.10000
    Component Sizing Information, Fan:ConstantVolume, THERMAL ZONE 1-3|VTU-1|SMPL_SUPPLYCONSTVOL, Design Size Maximum Flow Rate [m3/s], 1.30000
    Component Sizing Information, Coil:Cooling:DX:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1|VRF COOLING COIL, Design Size Rated Air Flow Rate [m3/s], 1.30000
    Component Sizing Information, Coil:Cooling:DX:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1|VRF COOLING COIL, Design Size Gross Rated Total Cooling Capacity [W], 21519.61596
    Component Sizing Information, Coil:Cooling:DX:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1|VRF COOLING COIL, Design Size Gross Rated Sensible Heat Ratio, 0.79866
    Component Sizing Information, Coil:Heating:DX:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1|VRF HEATING COIL, Design Size Rated Air Flow Rate [m3/s], 1.30000
    Component Sizing Information, Coil:Heating:DX:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1|VRF HEATING COIL, Design Size Gross Rated Heating Capacity [W], 21519.61596

**Comment by mjwitte** [comment]
Maybe I'm asking for too much - I guess I don't know if other ZoneHVAC honors the parent flow rate when sizing coils.  Hope you can fix it with a minimum number of nightcrawlers.

**Comment by rraustad** [comment]
Thinking about what should or shouldn't be done and using one input file for all geographies.

Autosize parent and child - sized the same, can be used in any geography
Hard size parent, autosize child - sized the same, values passed to child, can be used in any geography
Autosize parent, hard size child - sized differently, could run into problems in different geography
Hard size parent and child - you get what you get

So let the CI rip and see what happens. Note the size of the DX coils are still the same, I guess that could also be a problem for DX coils if air flows are too different.

    Component Sizing Information, Coil:Cooling:DX:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1|VRF COOLING COIL, Design Size Rated Air Flow Rate [m3/s], 1.30000
    Component Sizing Information, Coil:Cooling:DX:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1|VRF COOLING COIL, Design Size Gross Rated Total Cooling Capacity [W], 21519.61596
    Component Sizing Information, Coil:Cooling:DX:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1|VRF COOLING COIL, Design Size Gross Rated Sensible Heat Ratio, 0.79866
    Component Sizing Information, Coil:Heating:DX:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1|VRF HEATING COIL, Design Size Rated Air Flow Rate [m3/s], 1.20000
    Component Sizing Information, Coil:Heating:DX:VariableRefrigerantFlow, THERMAL ZONE 1-3|VTU-1|VRF HEATING COIL, Design Size Gross Rated Heating Capacity [W], 21519.61

**Comment by mjwitte** [comment]
Closed via #6121
